### PR TITLE
Setup CircleCI

### PR DESCRIPTION
**Pull request by dkubb**
This branch sets up CircleCI so that it builds each commit.

### CODE REVIEW

**Comment by dkubb** [comment]
@mbj this is ready for review.

**Comment by dkubb** [comment]
@mbj do you think I should change the `Gemfile` to reference our spree fork?

**Comment by mbj** [comment]
> @mbj do you think I should change the Gemfile to reference our spree fork?

I think it should reference our spree fork. This will make the builds much more relevant and reduce cycles on reviews.

**Comment by dkubb** [comment]
@mbj I've added a commit that updates the `Gemfile` to reference our spree fork.

When the build is green this is ready for review.

File: Gemfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 source 'https://rubygems.org'
 
-gem 'spree', github: 'spree/spree', branch: '2-3-stable'
+gem 'spree', github: 'MountainRoseHerbs/spree', branch: '2-3-stable-mrh'
```
Comment (mbj) [inline Gemfile]: Does this `github` option expand to the https URL or the plain text `git://` protocol? If its the latter I think we should stay with using the explicit HTTPS URL.

Comment (dkubb) [inline Gemfile]: Nice catch. Yes it does expand to the `git://` protocol.

The bundler people plan to change it to expand to `https://` when bundler 2.0 is released, but do not plan to change it during the 1.x release cycle.

**Comment by mbj** [comment]
Code wise I'm fine. Waiting for CI than will merge.

**Pull request merged**